pkg/console/controllers/managedclusters/controller_test.go

```diff
@@ -0,0 +1,122 @@
+
+	v1 "github.com/open-cluster-management/api/cluster/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/console-operator/pkg/console/controllers/util"
+)
+
+func generateClusters(t *testing.T) []v1.ManagedCluster {
+
+	clusters := []v1.ManagedCluster{}
+
+	cluster1 := v1.ManagedCluster{}
+	cluster1.ObjectMeta = metav1.ObjectMeta{Name: "ProductAKSVer412"}
+	cluster1.Status = v1.ManagedClusterStatus{
+		ClusterClaims: []v1.ManagedClusterClaim{
+			{Name: "product.open-cluster-management.io", Value: "AKS"},
+			{Name: "version.openshift.io", Value: "4.12.0-ec.1"},
+		},
+	}
+
+	clusters = append(clusters, cluster1)
+
+	cluster2 := v1.ManagedCluster{}
+	cluster2.ObjectMeta = metav1.ObjectMeta{Name: "ProductOpenshiftVer49"}
+	cluster2.Status = v1.ManagedClusterStatus{
+		ClusterClaims: []v1.ManagedClusterClaim{
+			{Name: "product.open-cluster-management.io", Value: "Openshift"},
+			{Name: "version.openshift.io", Value: "4.9.0-ec.1"},
+		},
+	}
+	clusters = append(clusters, cluster2)
+
+	cluster3 := v1.ManagedCluster{}
+	cluster3.ObjectMeta = metav1.ObjectMeta{Name: "ProductOpenshiftVer39"}
+	cluster3.Status = v1.ManagedClusterStatus{
+		ClusterClaims: []v1.ManagedClusterClaim{
+			{Name: "product.open-cluster-management.io", Value: "Openshift"},
+			{Name: "version.openshift.io", Value: "3.9"},
+		},
+	}
+	clusters = append(clusters, cluster3)
+
+	cluster4 := v1.ManagedCluster{}
+	cluster4.ObjectMeta = metav1.ObjectMeta{Name: "ProductEKSVer311"}
+	cluster4.Status = v1.ManagedClusterStatus{
+		ClusterClaims: []v1.ManagedClusterClaim{
+			{Name: "product.open-cluster-management.io", Value: "EKS"},
+			{Name: "version.openshift.io", Value: "3.11.rcbeata"},
+		},
+	}
+	clusters = append(clusters, cluster4)
+
+	return clusters
+}
+
+func TestGenerateManagedClusterController(t *testing.T) {
+	clusters := generateClusters(t)
+	t.Logf("\n>>>>>>> Validating %d clusters have been created\n...", len(clusters))
+}
+
+func TestProductCheckForClusterList(t *testing.T) {
+	expectedCluster := []string{}
+	expectedCluster = append(expectedCluster, "ProductOpenshiftVer49")
+	expectedCluster = append(expectedCluster, "ProductOpenshiftVer39")
+	clusters := generateClusters(t)
+	validClusters := []string{}
+	if len(clusters) < 1 {
+		t.Fatalf("\n---------------\nNo Valid clusters, count was %v\n-----------", len(clusters))
+	}
+	for _, managedCluster := range clusters {
+		clusterName := managedCluster.GetName()
+		claims := util.ClaimsMap(&managedCluster)
+
+		// Omit clusters that have the following product name
+		omittedProductCluster := []string{"AKS", "EKS", "IKS", "ROKS"}
+		if ProductCheckForClusterList(claims, omittedProductCluster) {
+			t.Logf(">>>>> Skipping managed cluster %q, product %q is unsupported", clusterName, claims["product.open-cluster-management.io"])
+			continue
+		}
+		validClusters = append(validClusters, clusterName)
+	}
+
+	if diff := deep.Equal(validClusters, expectedCluster); diff != nil {
+		t.Error(diff)
```

Comment:
When this function is called the program will fail, so no other calls will get executed.